File: Source/Gamemodes/Solo/Solo.cs
public static class Solo
{
    public static void Initialize()
    {
        ItemSpawner.NUMBER_OF_ITEMS = 8;
        ShadowKitty.Initialize();
    }

    public static void ReviveKittySoloTournament(Kitty kitty)
    {
        if (Gamemode.CurrentGameModeType != "Race") return; // Race GamemodeType.
        new SoloDeathTimer(kitty.Player);
    }

    public static void RoundEndCheck()
    {
        if (Gamemode.CurrentGameModeType != "Progression") return; // Progression mode

        for(int i = 0; i < Globals.ALL_PLAYERS.Count; i++)
        {
            var kitty = Globals.ALL_KITTIES[Globals.ALL_PLAYERS[i]];
            if (kitty.Alive) return;
        }
        RoundManager.RoundEnd();
    }
}

File: src/Gamemodes/Solo/Solo.ts
import { Kitty } from 'src/Game/Entities/Kitty/Kitty'
import { ItemSpawner } from 'src/Game/Items/ItemSpawner'
import { Globals } from 'src/Global/Globals'
import { CurrentGameMode } from '../CurrentGameMode'
import { GameMode } from '../GameModeEnum'
import { SoloDeathTimer } from './SoloDeathTimer'

export class Solo {
    public static Initialize = () => {
        ItemSpawner.NUMBER_OF_ITEMS = 8
    }

    public static ReviveKittySoloTournament = (kitty: Kitty) => {
        if (CurrentGameMode.active !== GameMode.SoloTournament || Globals.CurrentGameModeType !== 'Race') return // Solo Gamemode & Race GamemodeType.
        new SoloDeathTimer(kitty.Player)
    }
}
